mango-core/src/main/scala/org/bdgenomics/mango/models/GenotypeMaterialization.scala

```diff
@@ -79,6 +81,60 @@ class GenotypeMaterialization(s: SparkContext,
     // write variants and genotypes to json
     genotypes.mapValues(v => write(v))
   }
+
+  /**
+   * Formats raw data from RDD to JSON.
+   *
+   * @param region Region to obtain coverage for
+   * @param binning Tells what granularity of coverage to return. Used for large regions
+   * @return JSONified data map
+   */
+  def getGenotype(region: ReferenceRegion, binning: Int = 1): Map[String, String] = {
+    val data: RDD[(String, Genotype)] = get(region)
+    if (binning == 1) {
+      return stringify(data)
+    }
+    val flattened: Map[String, Array[(String, VariantJson)]] = data
+      .map(r => {
+        val geno = r._2
+        val bin: Int = (geno.getStart / binning).toInt
+        ((r._1, bin), geno)
+      })
+      .reduceByKey((a, b) => {
+        val end = max(a.getEnd, b.getEnd)
+        val start = max(a.getStart, b.getStart)
+        a.setStart(start)
+        a.setEnd(end)
+        a
+      })
+      .map(r => {
+        val n = r._2
+        var ref = n.getVariant.getReferenceAllele
+        var alt = n.getVariant.getAlternateAllele
+        if (n.getEnd - n.getStart != ref.length) {
+          ref = "NNNN"
```

Comment:
We need a placeholder to represent the ref and alternate alleles when we bin the variants. Is there something else that would work better here?